src/lib/concurrency/transaction_manager.cpp

```diff
@@ -23,7 +23,41 @@ TransactionManager::TransactionManager()
 CommitID TransactionManager::last_commit_id() const { return _last_commit_id; }
 
 std::shared_ptr<TransactionContext> TransactionManager::new_transaction_context() {
-  return std::make_shared<TransactionContext>(_next_transaction_id++, _last_commit_id);
+  std::unique_lock<std::mutex> lock(_mutex_active_snapshot_commit_ids);
+  const TransactionID snapshot_commit_id = _last_commit_id;
+  _active_snapshot_commit_ids.insert(snapshot_commit_id);
+  return std::make_shared<TransactionContext>(_next_transaction_id++, snapshot_commit_id);
+}
+
+void TransactionManager::deregister_transaction(CommitID snapshot_commit_id) {
+  std::unique_lock<std::mutex> lock(_mutex_active_snapshot_commit_ids);
+
+  auto it =
+      std::find(std::begin(_active_snapshot_commit_ids), std::end(_active_snapshot_commit_ids), snapshot_commit_id);
+
+  if (it != std::end(_active_snapshot_commit_ids)) {
+    _active_snapshot_commit_ids.erase(it);
+    return;
+  }
+
+  /**
+   * TODO(all) Change Google tests like get_table_test.cpp to always use the TransactionManager for creating
```

Comment:
Update: I tried to change the tests → https://github.com/d-justen/hyrise/issues/33#issuecomment-463806010

What do you suggest @mrks?

Reply:
Would it be possible to create *and register* these by hand?

Reply:
Yep, possible. I already started implementing this. Then, a thought came to my mind: Instead of calling `register_transaction()` manually at different places, we could also call it in the `TransactionContext`-constructor once.
In addition, we could make `TransactionContext` a friend class of `TransactionManager`. Then, `register_transaction()` and `deregister_transaction()` could be made private and thus removed from the public interface of `TransactionManager`.

Reply:
Sounds even better.